earth_enterprise/src/common/etencoder.cc

```diff
@@ -109,6 +109,10 @@ const std::string kDefaultKey(
 
 
 void Encode(void* data, uint32 datalen, const void* key, uint32 keylen) {
+  if (keylen % 8 != 0) {
+    throw khSimpleException("Encryption keys must be a multiple of 8 bytes long.");
```

Comment:
Should this have an empty return like the if block on lines 126-130?

Reply:
I interpreted the return on line 129 as being a part of the standard processing when the encoded string is the same as the input string, so it's not really an error condition. That's why I opted to throw an exception here to make it clear that there was a problem, not normal processing. I guess you could argue line 129 is also error checking since it checks for null pointers as well, but I think it's safe to say that when this condition occurs we want processing to stop, not go on like normal.